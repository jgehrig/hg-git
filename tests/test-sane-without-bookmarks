#!/bin/sh

# Fails for some reason, need to investigate
# "$TESTDIR/hghave" git || exit 80

# bail early if the user is already running git-daemon
echo hi | nc localhost 9418 && exit 80

echo "[extensions]" >> $HGRCPATH
echo "hggit=$(echo $(dirname $(dirname $0)))" >> $HGRCPATH
echo 'hgext.graphlog =' >> $HGRCPATH

GIT_AUTHOR_NAME='test'; export GIT_AUTHOR_NAME
GIT_AUTHOR_EMAIL='test@example.org'; export GIT_AUTHOR_EMAIL
GIT_AUTHOR_DATE="2007-01-01 00:00:00 +0000"; export GIT_AUTHOR_DATE
GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"; export GIT_COMMITTER_NAME
GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"; export GIT_COMMITTER_EMAIL
GIT_COMMITTER_DATE="$GIT_AUTHOR_DATE"; export GIT_COMMITTER_DATE

count=10
commit()
{
    GIT_AUTHOR_DATE="2007-01-01 00:00:$count +0000"
    GIT_COMMITTER_DATE="$GIT_AUTHOR_DATE"
    git commit "$@" >/dev/null 2>/dev/null || echo "git commit error"
    count=`expr $count + 1`
}

mkdir gitrepo
cd gitrepo
git init | python -c "import sys; print sys.stdin.read().replace('$(dirname $(pwd))/', '')"
echo alpha > alpha
git add alpha
commit -m 'add alpha'
echo beta > beta
git add beta
commit -m 'add beta'

# dulwich does not presently support local git repos, workaround
cd ..
git daemon --base-path="$(pwd)"\
 --listen=127.0.0.1\
 --export-all\
  --pid-file=gitdaemon.pid \
 --detach --reuseaddr

hg gclone git://127.0.0.1/gitrepo hgrepo
cd hgrepo
hg log --graph

cd ..
kill `cat gitdaemon.pid`
