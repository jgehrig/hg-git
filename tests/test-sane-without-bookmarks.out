Initialized empty Git repository in gitrepo/.git/

creating bookmarks failed, do you have bookmarks enabled?
importing git objects into hg
2 files updated, 0 files merged, 0 files removed, 0 files unresolved
@  changeset:   1:7bcd915dc873
|  tag:         default/master
|  tag:         tip
|  user:        test <test@example.org>
|  date:        Mon Jan 01 00:00:11 2007 +0000
|  summary:     add beta
|
o  changeset:   0:3442585be8a6
   user:        test <test@example.org>
   date:        Mon Jan 01 00:00:10 2007 +0000
   summary:     add alpha

